### ISSUE TITLE
Crash with self-hosted Posthog and session recordings

### ISSUE THREAD

**Issue by nessup** [PostHog/posthog-js#724]
When I use Posthog 1.68.5 with self-hosted Posthog (1.43.0), I get the following crash on the server side:

```
Traceback (most recent call last):
  File \"/python-runtime/django/core/handlers/exception.py\", line 47, in inner
    response = get_response(request)
  File \"/python-runtime/django/core/handlers/base.py\", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File \"/python-runtime/django/views/decorators/csrf.py\", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File \"/code/posthog/logging/timing.py\", line 13, in wrapper
    return func(*args, **kwargs)
  File \"/code/posthog/api/capture.py\", line 252, in get_event
    events = preprocess_session_recording_events_for_clickhouse(events)
  File \"/code/posthog/helpers/session_recording.py\", line 145, in preprocess_session_recording_events_for_clickhouse
    result.extend(list(compress_and_chunk_snapshots(snapshots)))
  File \"/code/posthog/helpers/session_recording.py\", line 153, in compress_and_chunk_snapshots
    has_full_snapshot = any(snapshot_data[\"type\"] == RRWEB_MAP_EVENT_TYPE.FullSnapshot for snapshot_data in data_list)
  File \"/code/posthog/helpers/session_recording.py\", line 153, in <genexpr>
    has_full_snapshot = any(snapshot_data[\"type\"] == RRWEB_MAP_EVENT_TYPE.FullSnapshot for snapshot_data in data_list)
TypeError: list indices must be integers or slices, not str
```

I also get this error in the logs:
```
Failed to decompress data. Compressed file ended before the end-of-stream marker was reached
```

I understand self-hosted Posthog is deprecated. We could certainly get around this issue by pinning our version of posthog-js to a version before the data format of session recordings changed, but obviously that comes with security implications.

Is there any workaround?

**Comment by BEGEMOT9I** [comment]
Yeah, we have the same problem, empirically it turned out that the problem is in version 1.68.2 and in this pull request: https://github.com/PostHog/posthog-js/commit/e7af347cb77e4277c97a880c32b3389d3c4cb18f

**Comment by pauldambra** [comment]
Hey,

Yeah, you'd need to either update the self-hosted version or downgrade posthog-js. Using 1.43.0 (although 1.43.1 is available too) suggests you're still using kubernetes? I'll highlight this in case you missed it https://posthog.com/blog/sunsetting-helm-support-posthog

**Comment by nessup** [comment]
@pauldambra Did that work for you? I tried that and did not see a difference

**Comment by BEGEMOT9I** [comment]
Yep, there are no such changes in the last release for 2 months now

**Comment by eriksobolt** [comment]
I got this same error as well. I have just installed posthog on a vm so no kubernetes.

i think what is confusing is the following line in the docs `You'll now be asked to provide the release tag you would like to use, as well as the domain you have connected to your instance.`

I just referred to the latest release tag I could find (1.43.1) and am running that. However your blogpost @pauldambra refers to `Finally, there will no longer be numbered releases of PostHog. Instead, we'll build a Docker image for each commit that happens in our GitHub repo. This means folks using our Docker Compose deployment will benefit immediately from new features and bugfixes, just like those on PostHog Cloud. You'll also be able to pin a version or roll back where necessary.`

So it will be all resolved by running the latest version, which is fine. But maybe this should be made clearer in the docs
